api/src/main/java/ai/djl/nn/recurrent/RecurrentBlock.java

```diff
@@ -227,22 +227,24 @@ protected NDList opInputs(ParameterStore parameterStore, NDList inputs, boolean
         long batchSize = inputs.head().getShape().get(0);
         inputs = updateInputLayoutToTNC(inputs);
         NDArray head = inputs.head();
+        NDManager manager = head.getManager();
         Device device = head.getDevice();
 
         NDList result = new NDList(head);
         try (NDList parameterList = new NDList()) {
             for (Parameter parameter : parameters.values()) {
-                NDArray array = parameterStore.getValue(parameter, device, training);
+                NDArray array = parameterStore.getValue(parameter, device, training).duplicate();
```

Comment:
I don't think we need dup it,
We can just attach flattened array to manager.
```
                NDArray array = parameterStore.getValue(parameter, device, training).flatten();
                array.attach(manager);
                parameterList.add(array);
```

Reply:
You're right, changed.